multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/listeners/EnterTestingPhaseListener.java

```diff
@@ -1,17 +1,31 @@
 package org.cloudfoundry.multiapps.controller.process.listeners;
 
+import javax.inject.Inject;
 import javax.inject.Named;
 
+import org.cloudfoundry.multiapps.controller.api.model.Operation;
+import org.cloudfoundry.multiapps.controller.persistence.services.OperationService;
 import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.variables.VariableHandling;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
 import org.flowable.engine.delegate.DelegateExecution;
 
 @Named("enterTestingPhaseListener")
 public class EnterTestingPhaseListener extends AbstractProcessExecutionListener {
 
     private static final long serialVersionUID = 1L;
 
+    private final OperationService operationService;
+
+    @Inject
+    public EnterTestingPhaseListener(OperationService operationService) {
+        this.operationService = operationService;
+    }
+
     @Override
     protected void notifyInternal(DelegateExecution execution) {
+        String correlationId = VariableHandling.get(execution, Variables.CORRELATION_ID);
```

Comment:
Should be checked that "correlation id" is always passed in every call actity/sub process where this listener is used.